core/src/test/scala/org/apache/spark/deploy/master/MasterSuite.scala

```diff
@@ -704,6 +707,43 @@ class MasterSuite extends SparkFunSuite
   private def getState(master: Master): RecoveryState.Value = {
     master.invokePrivate(_state())
   }
+
+  test("Total cores is not divisible by cores per executor") {
```

Comment:
I see, if there's no better way to verify it, I think it is not useful to add this two UTs.